### PR TITLE
[SPARK-44534][K8S] Handle only shuffle files in KubernetesLocalDiskShuffleExecutorComponents

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to prevent any future regressions due to new block types by checking `isShuffle` explicitly before invoking `BlockManager.TempFileBasedBlockStoreUpdater` in `KubernetesLocalDiskShuffleExecutorComponents`. `isShuffle` allows only 4 block types currently.

https://github.com/apache/spark/blob/58e5d86cc076d4546dac5e1f594977d615ec1e7a/core/src/main/scala/org/apache/spark/storage/BlockId.scala#L43-L46

### Why are the changes needed?

We want to protect the logic from non-shuffle blocks and new future type blocks.

```
$ k logs group-by-test-exec-6 | grep 'Try to recover'
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover shuffle data.
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/11/broadcast_0_piece0
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/11/shuffle_0_3136_0.data
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/11/shuffle_0_3312_0.data
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/11/shuffle_0_3402_0.data
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/11/shuffle_0_3752_0.data
23/07/24 22:59:45 INFO KubernetesLocalDiskShuffleExecutorComponents: Try to recover 
/data/spark-1/executor-x/blockmgr-716520a6-2a94-4418-8a23-d56fd9f4191b/0e/broadcast_0
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs with the newly added test case.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @viirya . Could you review this PR when you have some time?

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala [hunk 1]
```diff
@@ -98,8 +98,11 @@ object KubernetesLocalDiskShuffleExecutorComponents extends Logging {
       logInfo(s"Try to recover ${f.getAbsolutePath}")
       try {
         val id = BlockId(f.getName)
-        val decryptedSize = f.length()
-        bm.TempFileBasedBlockStoreUpdater(id, level, classTag, f, decryptedSize).save()
+        // To make it sure to handle only shuffle blocks
+        if (id.isShuffle) {
+          val decryptedSize = f.length()
+          bm.TempFileBasedBlockStoreUpdater(id, level, classTag, f, decryptedSize).save()
+        }
```
Comment (viirya) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala:105]: Do we need to log it? For example we did log for `UnrecognizedBlockId` below.

**Review by viirya — APPROVED** [review]

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala:105]: Thank you for review. Sure, let me add a log with `else` statement.

**Review by yaooqinn — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya and @yaooqinn . 
Merged to master/3.5.

**Pull request closed without merging**